source/fe/mapping_q_generic.cc

```diff
@@ -817,18 +796,13 @@ MappingQGeneric<dim, spacedim>::InternalData::initialize_face(
 
   if (dim > 1 && tensor_product_quadrature)
     {
-      const unsigned int  facedim = dim > 1 ? dim - 1 : 1;
-      const FE_Q<facedim> fe(polynomial_degree);
+      constexpr unsigned int facedim = dim > 1 ? dim - 1 : 1;
```

Comment:
Since we already assume `dim > 1` in the preceding if statement, could we get rid of this conditional while we are here?

Reply:
I removed this - thanks to the changes in #9829 where I also created instantiations for the `dim=0` case.